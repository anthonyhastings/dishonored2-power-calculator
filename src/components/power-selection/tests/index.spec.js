import renderer from 'react-test-renderer';
import PowerSelection from '../';

jest.mock('Components/avatar', () => 'MockAvatar');

jest.mock('Components/button', () => 'MockButton');

jest.mock('../powers-list', () => 'MockPowersList');

describe('PowerSelection component', () => {
  let testContext;

  const renderComponent = (props = {}) => (
    <PowerSelection {...testContext.defaultProps} {...props} />
  );

  beforeEach(() => {
    testContext = {};

    testContext.defaultProps = {
      character: { name: 'Mock Name', slug: 'mock-slug' },
      topLevelEnhancements: ['mock-top-level-enhancements'],
      topLevelPowers: ['mock-top-level-powers'],
    };
  });

  describe('when rendered', () => {
    beforeEach(() => {
      testContext.component = renderer.create(renderComponent());
    });

    it('renders a header area and categories for powers and enhancements', () => {
      expect(testContext.component).toMatchSnapshot();
    });
  });
});
