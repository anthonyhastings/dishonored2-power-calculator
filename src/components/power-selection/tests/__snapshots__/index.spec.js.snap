// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PowerSelection component when rendered renders a header area and categories for powers and enhancements 1`] = `
<section
  className="power-selection"
>
  <header
    className="power-selection__header"
  >
    <h1
      className="power-selection__title"
    >
      Mock Name
      <br />
      Select your powers
    </h1>
    <MockAvatar
      className="power-selection__avatar"
      name="Mock Name"
      slug="mock-slug"
    />
  </header>
  <div
    className="power-selection__categories"
  >
    <MockPowersList
      className="power-selection__category"
      powers={
        Immutable.List [
          "mock-top-level-powers",
        ]
      }
    >
      Top-level Powers
    </MockPowersList>
    <MockPowersList
      className="power-selection__category"
      powers={
        Immutable.List [
          "mock-top-level-enhancements",
        ]
      }
    >
      Top-level Enhancements
    </MockPowersList>
  </div>
  <footer
    className="power-selection__footer"
  >
    <MockButton
      className="power-selection__back-cta"
      href="/"
    >
      Back to character selection
    </MockButton>
  </footer>
</section>
`;
